e2e/test/iothub/service/RegistryManagerExportDevicesTests.cs

```diff
@@ -163,55 +155,57 @@ public async Task RegistryManager_ExportDevices(StorageAuthenticationType storag
             RegistryManager registryManager,
             Uri containerUri)
         {
-            int tryCount = 0;
-
             ManagedIdentity identity = isUserAssignedMsi
                 ? new ManagedIdentity
                 {
                     UserAssignedIdentity = TestConfiguration.IoTHub.UserAssignedMsiResourceId
                 }
                 : null;
 
-            JobProperties exportJobResponse = JobProperties.CreateForExportJob(
+            JobProperties jobProperties = JobProperties.CreateForExportJob(
                 containerUri.ToString(),
                 true,
                 devicesFileName,
                 storageAuthenticationType,
                 identity);
-            exportJobResponse.IncludeConfigurations = true;
-            exportJobResponse.ConfigurationsBlobName = configsFileName;
+            jobProperties.IncludeConfigurations = true;
+            jobProperties.ConfigurationsBlobName = configsFileName;
 
-            while (tryCount < MaxIterationWait)
+            var sw = Stopwatch.StartNew();
+
+            while (!jobProperties.IsFinished)
             {
                 try
                 {
-                    exportJobResponse = await registryManager.ExportDevicesAsync(exportJobResponse).ConfigureAwait(false);
+                    jobProperties = await registryManager.ExportDevicesAsync(jobProperties).ConfigureAwait(false);
+                    if (!string.IsNullOrWhiteSpace(jobProperties.FailureReason))
+                    {
+                        Logger.Trace($"Job failed due to {jobProperties.FailureReason}");
+                    }
                     break;
                 }
                 // Concurrent jobs can be rejected, so implement a retry mechanism to handle conflicts with other tests
-                catch (JobQuotaExceededException) when (++tryCount < MaxIterationWait)
+                catch (JobQuotaExceededException)
                 {
-                    Logger.Trace($"JobQuotaExceededException... waiting.");
+                    Logger.Trace($"JobQuotaExceededException... waiting after {sw.Elapsed}.");
                     await Task.Delay(s_waitDuration).ConfigureAwait(false);
                     continue;
                 }
             }
 
-            for (int i = 0; i < MaxIterationWait; ++i)
+            sw.Stop();
+            Logger.Trace($"Job started after {sw.Elapsed}.");
```

Comment:
Print out some time taken to start the job or wait for it to finish so we get an idea of how long we're waiting if a test times out.